specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,280 @@
+
+    For an example, see the introduction to the PasswordBox control
+    (http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box).
+-->
+
+NotifyDropTargetAction is meant to provide a way for composition hosted WebViews to receive drop events as part of a drag/drop operation.
+It is the hosting application's responsibility to call RegisterDragDrop (https://docs.microsoft.com/en-us/windows/win32/api/ole2/nf-ole2-registerdragdrop)
+on the HWND that contains any composition hosted WebViews and to implement IDropTarget(https://docs.microsoft.com/en-us/windows/win32/api/oleidl/nn-oleidl-idroptarget)
+to receive the corresponding drop events from Ole32:
+
+IDropTarget::DragEnter
+IDropTarget::DragMove
+IDropTarget::DragLeave
+IDropTarget::Drop
+
+NotifyDropTargetAction consolidates these four functions into one and the COREWEBVIEW2_DROP_TARGET_ACTION enum
+specifies which correspondingIDropTarget function was called.
+
+
+# Examples
+<!-- TEMPLATE
+    Use this section to explain the features of the API, showing
+    example code with each description in both C# (for our WinRT API or .NET API) and
+    in C++ for our COM API. Use snippets of the sample code you wrote for the sample apps.
+
+    The general format is:
+
+    ## FirstFeatureName
+
+    Feature explanation text goes here, including why an app would use it, how it
+    replaces or supplements existing functionality.
+
+    ```c#
+    void SampleMethod()
+    {
+        var show = new AnExampleOf();
+        show.SomeMembers = AndWhyItMight(be, interesting)
+    }
+    ```
+    
+    ```cpp
+    void SampleClass::SampleMethod()
+    {
+        winrt::com_ptr<ICoreWebView2> webview2 = ...
+    }
+    ```
+
+    ## SecondFeatureName
+
+    Feature explanation text goes here, including why an app would use it, how it
+    replaces or supplements existing functionality.
+
+    ```c#
+    void SampleMethod()
+    {
+        var show = new AnExampleOf();
+        show.SomeMembers = AndWhyItMight(be, interesting)
+    }
+    ```
+    
+    ```cpp
+    void SampleClass::SampleMethod()
+    {
+        winrt::com_ptr<ICoreWebView2> webview2 = ...
+    }
+    ```
+
+    As an example of this section, see the Examples section for the PasswordBox
+    control (https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). 
+-->
+
+```c++
+// Initialized elsewhere
+wil::com_ptr<ICoreWebView2Controller> webViewController;
+
+webViewCompositionController =
+        webViewController.query<ICoreWebView2CompositionController>();
+
+// Implementation for IDropTarget
+HRESULT DropTarget::DragEnter(IDataObject * data_object,
+    DWORD key_state,
+    POINTL cursor_position,
+    DWORD* effect)
+{
+    POINT point = {cursor_position.x, cursor_position.y};
+    // Tell the helper that we entered so it can update the drag image and return
+    // the allowed effects.
+    IDropTargetHelper* drop_helper = DropHelper();
```

Comment:
Is DropHelper a method we define? I'm wondering why you don't need to use a com_ptr for the result of it. Does that get the same one DropHelper each time? Maybe GetDropHelper() would be clearer?